core/bloombits/generator.go

```diff
@@ -80,8 +80,8 @@ func (b *Generator) Bitset(idx uint) ([]byte, error) {
 	if b.nextBit != b.sections {
 		return nil, errors.New("bloom not fully generated yet")
 	}
-	if idx >= b.sections {
-		return nil, errSectionOutOfBounds
+	if idx >= types.BloomBitLength {
```

Comment:
Are you sure this change is correct? We have an awful lof of `b.nextBit >= b.sections` comparisons, so if those are good, shouldn't here the original code be good too? Could you give an example of where this goes wrong?

Reply:
Here `b.nextBit` actually represents the `nextSection`. While for the `Bitset` function, the index represents bit index.

Reply:
Indeed, I stand corrected!